### ISSUE TITLE
Saving the code is not formatting anymore in 1.11.0/1

### ISSUE THREAD

**Issue by vikas027** [Microsoft/vscode#24176]
- VSCode Version: 1.11.1
- OS Version: OS X El Capitan (10.11.2)

In previous version, when I saved the file (cmnd+s) automagically fixed the formatting for me, but it is not happening anymore.

Steps to Reproduce:

1. Install VSCode 1.11.1 on OS X 10.11.2 
2. Create a code without formatting
```
  tag {
    key     = "terraform"
    value          = true
    propagate_at_launch = true
  }
```
3. Try Saving the file (cmnd+s)

**Comment by chrmarti** [comment]
What language are you formatting? Which extensions do you have installed (`code --list-extensions`)?

**Comment by vikas027** [comment]
Hey @chrmarti ,

I am using Terraform ([HCL](https://github.com/hashicorp/hcl)) code. Before the upgrade, I was able to save the code and it used to format everything for me automagically.

Here is the list.
```
🍺  /Applications/Visual Studio Code.app/Contents/Resources/app/bin$ ./code --list-extensions
JerryHong.autofilename
PeterJausovec.vscode-docker
RoscoP.ActiveFileInStatusBar
alefragnani.project-manager
alexdima.copy-relative-path
donjayamanne.githistory
donjayamanne.listFiles
eamodio.gitlens
emilast.LogFileHighlighter
felipecaputo.git-project-manager
gizak.shortcuts
groksrc.ruby
haaaad.ansible
jtlowe.vscode-icon-theme
marcostazi.VS-code-vagrantfile
mauve.terraform
misogi.ruby-rubocop
mohsen1.prettify-json
ms-vscode.sublime-keybindings
mushan.vscode-open-iterm2
naumovs.color-highlight
qinjia.view-in-browser
rebornix.Ruby
robertohuertasm.vscode-icons
secanis.jenkinsfile-support
timonwong.ansible-autocomplete
tomoki1207.pdf
waderyan.gitblame
wholroyd.HCL
wholroyd.jinja
ybaumes.highlight-trailing-white-spaces
🍺  /Applications/Visual Studio Code.app/Contents/Resources/app/bin$
```

**Comment by vikas027** [comment]
Hello @chrmarti ,

I have found the cause of the issue. Somehow after the upgrade, my terraform files (`*.tf`) got mapped to `HCL` files. I had to fix this, by placing below lines in `WORKSPACE SETTINGS`

```json
{
    "files.associations": {
        "*.tf": "terraform",
        "*.tfstate": "json"
    }
}
```

My code was formatted automatically on clicking save as I have this in my `settings.json`
```json
  // Run 'terraform fmt' on save.
  "terraform.formatOnSave": true,
```

Closing this issue as it is not related to VSCode, apologies for the unwanted noise. I am fairly new to VSCode; switched from Sublime Text recently.

Cheers.
